libs/cli/src/generators/ui/generator.ts

```diff
@@ -1,11 +1,13 @@
 import { type GeneratorCallback, type Tree, addDependenciesToPackageJson, runTasksInSerial } from '@nx/devkit';
 import { prompt } from 'enquirer';
+import { Config, getOrCreateConfig } from '../../utils/config';
 import { SPARTAN_COLLAPSIBLE_BRAIN_VERSION } from '../base/versions';
 import { addDependentPrimitives } from './add-dependent-primitive';
 import type { HlmUIGeneratorSchema } from './schema';
 
 export default async function hlmUIGenerator(tree: Tree, options: HlmUIGeneratorSchema & { angularCli?: boolean }) {
 	const tasks: GeneratorCallback[] = [];
+	const config = await getOrCreateConfig(tree);
```

Comment:
It may be confusing, when i set the --directory param, and then have to specify the directory again. 
I know it is just the first time, if the components.json does not exist. 
Maybe we can skip that step if the directory is set through options.

Reply:
That's right!
If we think config will have more fields we shouldn't skip this entirely but maybe only skip the step where we ask the directory.

What do you want for now when --directory is provided and config isn't there yet? A couple of options:
A) We entirely skip the config generation (this must be changed when we add more options in the config)
or
B) We try to create the config anyway, but instead of asking the directory we use the provided value from the params (this means today we skip showing prompts entirely, but when we'll add one more no other changes will be required)

Reply:
Oh, that is a valid point. 
I'd vote for B

Reply:
I see enquirer (the prompt library) has a `skip` property which also picks the `initial` value if skip is true, exactly what we needed! :)